pkg/chunk/chunk_store.go

```diff
@@ -141,11 +141,7 @@ func (c *store) PutOne(ctx context.Context, from, through model.Time, chunk Chun
 func (c *store) calculateIndexEntries(userID string, from, through model.Time, chunk Chunk) (WriteBatch, error) {
 	seenIndexEntries := map[string]struct{}{}
 
-	metricName, err := extract.MetricNameFromMetric(chunk.Metric)
-	if err != nil {
-		return nil, err
-	}
-
+	metricName := chunk.Metric.Get(labels.MetricName)
```

Comment:
This would fail if there was no metric name - should we preserve this?  I'm not sure TBH - the empty string could be considered a valid metric name...

Reply:
How does this relate to #1214, which suggests (a) we already enforce that everything has a metric name in Cortex and (b) in other situations you don't have a metric name.

Reply:
I added back the check.